client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,198 @@
+const getSiteNameFromURL = ( url: string ) => url?.split( '.' )?.[ 0 ] ?? url;
+
+function CalypsoWrappedMarketplaceDomainUpsell(): JSX.Element {
+	const [ selectedDomainProductUUID, setDomainProductUUID ] = useState< string >( '' );
+	const [ selectedDomain, setDomain ] = useState< DomainSuggestions.DomainSuggestion | null >(
+		null
+	);
+
+	const [ isExpandedBasketView, setIsExpandedBasketView ] = useState( false );
+	const { addProductsToCart, replaceProductsInCart, removeProductFromCart } = useShoppingCart();
+	const products = useSelector( getProductsList );
+	const previousPath = useSelector( getPreviousPath );
+	const isFetchingProducts = useSelector( isProductsListFetching );
+	const selectedSite = useSelector( getSelectedSite );
+	const { domain } = useSelector( ( state ) =>
+		getWpComDomainBySiteId( state, selectedSite && selectedSite.ID )
+	);
+	const siteName = getSiteNameFromURL( domain );
+	const dispatch = useDispatch();
+
+	useEffect( () => {
+		setIsExpandedBasketView( isDesktop() );
+		selectedSite && dispatch( fetchSiteDomains( selectedSite.ID ) );
+	}, [ setIsExpandedBasketView, selectedSite, dispatch ] );
+
+	useEffect( () => {
+		//FIXME: This code segment simulates yoast premium already being added when arriving here. To be removed when plugins page is completed.
+		if ( ! isFetchingProducts && products[ 'yoast_premium' ] ) {
+			const yoastProduct = fillInSingleCartItemAttributes(
+				{ product_slug: 'yoast_premium' },
+				products
+			);
+			replaceProductsInCart( [ yoastProduct ] );
+		}
+	}, [ isFetchingProducts, products, replaceProductsInCart ] );
+
+	const redirectToUseDomainFlow = (): void => {
+		const currentUrl = '/plugins/domain';
+		const useYourDomainUrl = `/start/new-launch/domains-launch/use-your-domain?siteSlug=${ selectedSite?.slug }&source=${ currentUrl }`;
+		page( useYourDomainUrl );
+	};
+
+	const onDomainSelect = async ( suggestion: DomainSuggestions.DomainSuggestion ) => {
+		const { product_slug, domain_name } = suggestion;
+		const domainProduct = {
+			...domainRegistration( {
+				productSlug: product_slug,
+				domain: domain_name,
+				source: 'Marketplace-Yoast-Domain-Upsell',
+			} ),
+			...suggestion,
+		};
+		setDomain( domainProduct );
+
+		//First remove the previously added domain
+		if ( selectedDomainProductUUID ) {
+			await removeProductFromCart( selectedDomainProductUUID );
+			setDomainProductUUID( '' );
+		}
+
+		//Then add the new domain
+		const responseCart = await addProductsToCart( [ domainProduct ] );
+		const productAdded = responseCart.products.find(
+			( { product_slug: added_product_slug } ) => added_product_slug === product_slug
+		);
+		setDomainProductUUID( productAdded?.uuid ?? '' );
+	};
+
+	return (
+		<>
+			<Masterbar>
+				<Item
+					url="#"
+					icon="cross"
+					className="marketplace-domain-upsell__close-button"
+					onClick={ () =>
+						previousPath
+							? page( previousPath )
+							: page( `/plugins/wordpress-seo/${ selectedSite?.slug }` )
```

Comment:
I think that if there's no `selectedSite`, we should remove the trailing slash.

Reply:
I am curious to learn what the effect of having this trailing slash in the world of pagejs?

Reply:
Page.js will have the user get redirected to the same path, but without the trailing slash. But why the extra redirect (or history state push) when we can avoid it? 😉

Reply:
Well ... to have lesser code 😁 
So to clarify, if there is a slash does it add an additional entry to the history stack? That seems like a weird behaviour 🤔  Or maybe I am not understanding this correctly. Any resources to read up more on this?

Reply:
Actually, just checked, it doesn't add an entry to the history, which is nice. Page.js [handles this internally](https://github.com/visionmedia/page.js/blob/master/page.js#L353). It still performs a change to the route, though, which is unwanted IMHO. And while I love your motivation to keep the code less (no code is even better because no bugs 🤣 ), I'd prefer to avoid changing the URL unnecessarily.

Reply:
Thanks for this explanation @tyxla ! Sorry for my pesky curiosity! I am fully hyped and learning a lot the past few days 😀 .  I will fix this up and get pack! 👍

Reply:
Please keep being curious! Curiosity feeds us all with knowledge 😋